src/main/java/frc/robot/commands/DriveWithJoysticks.java

```diff
@@ -0,0 +1,85 @@
+package frc.robot.commands;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
+import com.ctre.phoenix6.mechanisms.swerve.SwerveRequest;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.drive.CommandSwerveDrivetrain;
+import frc.robot.Constants.DriveConstants;
+import frc.robot.Utils.Deadband;
+
+public class DriveWithJoysticks extends Command {
+    CommandSwerveDrivetrain drivetrain;
+    DoubleSupplier x;
+    DoubleSupplier y;
+    DoubleSupplier rot;
+    BooleanSupplier fieldCentric;
+    BooleanSupplier babyMode;
+
+    double xMpS;
+    double yMpS;
+    double rotRadpS;
+
+    SwerveRequest.FieldCentric drive = new SwerveRequest.FieldCentric();
+    SwerveRequest.RobotCentric driveRobot = new SwerveRequest.RobotCentric();
+    SwerveRequest.SwerveDriveBrake brake = new SwerveRequest.SwerveDriveBrake();
+    SwerveRequest.PointWheelsAt point = new SwerveRequest.PointWheelsAt();
+
+    public DriveWithJoysticks(CommandSwerveDrivetrain drivetrain, DoubleSupplier x, DoubleSupplier y,
+            DoubleSupplier rot, BooleanSupplier fieldCentric, BooleanSupplier babyMode) {
+        this.drivetrain = drivetrain;
+        this.x = x;
+        this.y = y;
+        this.rot = rot;
+        this.fieldCentric = fieldCentric;
+        this.babyMode = babyMode;
+
+        addRequirements(drivetrain);
+    }
+
+    @Override
+    public void initialize() {
+    }
+
+    @Override
+    public void execute() {
+        double[] joystickInputsFiltered = Deadband.twoAxisDeadband(x.getAsDouble(), y.getAsDouble(),
+                DriveConstants.deadbandPercent);
+
+        xMpS = joystickInputsFiltered[0] * DriveConstants.MaxSpeedMetPerSec;
+        yMpS = joystickInputsFiltered[1] * DriveConstants.MaxSpeedMetPerSec;
+        rotRadpS = Deadband.oneAxisDeadband(rot.getAsDouble(), DriveConstants.deadbandPercent)
+                * DriveConstants.MaxAngularRateRadiansPerSec;
+
+        if (babyMode.getAsBoolean()) {
+            xMpS *= 0.5;
+            yMpS *= 0.5;
+            rotRadpS *= 0.5;
+        }
+
+        if (xMpS == 0 && yMpS == 0 && rotRadpS == 0) {
+            // If not moving, brake
+            drivetrain.setControl(brake);
+        } else if (fieldCentric.getAsBoolean()) {
+            drivetrain.setControl(drive.withVelocityX(-xMpS)
+                    .withVelocityY(-yMpS)
+                    .withRotationalRate(-rotRadpS));
+        } else {
+            drivetrain.setControl(driveRobot.withVelocityX(-xMpS)
+                    .withVelocityY(-yMpS)
+                    .withRotationalRate(-rotRadpS));
```

Comment:
If possible I'd prefer that we put `setControl` calls in `periodic` for the drivetrain and then use one or more setters to set the raw control values in this command